templates/ecommerce/billing-form-fields.php

```diff
@@ -24,6 +24,23 @@
 $billing_state      = $billing_info->billing_state ?? '';
 $billing_city       = $billing_info->billing_city ?? '';
 
+/**
+ * Retain form data when validation errors occur during guest checkout.
+ *
+ * @since 3.4.0
+ */
+if ( ! is_user_logged_in() ) {
```

Comment:
Instead of rewriting these variables, can't we use the input_old as the fallback for the above variables? For example:
`$billing_first_name = $billing_info->billing_first_name ?? tutor_utils()->input_old( 'billing_first_name', '' );`